flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkConfigBuilder.java

```diff
@@ -94,6 +97,17 @@ public FlinkConfigBuilder applyFlinkConfiguration() {
                     REST_SERVICE_EXPOSED_TYPE,
                     KubernetesConfigOptions.ServiceExposedType.ClusterIP);
         }
+
+        // With last-state upgrade mode, set the default value of 'execution.checkpointing.interval'
+        // to 5 minutes when HA is enabled.
+        if (spec.getJob() != null
+                && spec.getJob().getUpgradeMode() == UpgradeMode.LAST_STATE
+                && FlinkUtils.isKubernetesHAActivated(effectiveConfig)
```

Comment:
The HA config is already validated for last-state mode by the DeploymentValidator, no need to check again here